### PR TITLE
Support default exports.

### PR DESCRIPTION
**Pull request by mskelton**
Allows importing TypeScript files using `export default {`.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import { BackendModule, ReadCallback, Services } from 'i18next'
 
-type ResourceFetcher = () => Promise<Record<string, string>>
+type ResourceFetcher = () => Promise<{
+  __esModule?: true
```
Comment (avo) [inline src/index.ts:4]: Wondering on the double underscore?

Comment (avo) [reply src/index.ts:4]: That it's global vs just private?

Comment (avo) [reply src/index.ts:4]: Is our usage consistent?

**Review by avo — APPROVED** [review]

File: src/index.ts (hunk 1 above)
Comment (mskelton) [reply src/index.ts:4]: This just represents how Babel/TypeScript transpile code:

- https://www.typescriptlang.org/tsconfig#esModuleInterop
- https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs#example

**Pull request merged**